test/unit/alphabet/aminoacid/aminoacid_test_template.hpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/alphabet/aminoacid/all.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+template <typename T>
+class aminoacid : public ::testing::Test
+{};
+
+TYPED_TEST_CASE_P(aminoacid);
+
+TYPED_TEST_P(aminoacid, global_assign_char)
+{
+    using t = TypeParam;
+    std::vector<char> input
+    {
+        'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+        'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+        'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+        'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+        '*', '!'
+    };
+    std::vector<TypeParam> cmp;
+    if constexpr (std::is_same_v<TypeParam, aa27>)
```

Comment:
Remove the aa27/aa20 specific `input` and `cmp` and ut them into the corresponding single tests.